src/support.cpp

```diff
@@ -1479,30 +1479,30 @@ std::pair<Polygons, Polygons> AreaSupport::computeBasicAndFullOverhang(const Sli
     const Polygons outlines = mesh.layers[layer_idx].getOutlines();
     constexpr bool no_support = false;
     constexpr bool no_prime_tower = false;
-    const Polygons outlines_below = storage.getLayerOutlines(layer_idx - 1, no_support, no_prime_tower);
 
-    // To avoids generating support for textures on vertical surfaces, a moving average
-    // is taken over smooth_height. The smooth_height is currently an educated guess
-    // that we might want to expose to the frontend in the future.
     constexpr double smooth_height = 0.4; //mm
-    const auto layers_above = static_cast<LayerIndex>(std::round(smooth_height / mesh.settings.get<double>("layer_height")));
-    Polygons outlines_above;
-    for (int i = 0; layer_idx + i < mesh.layers.size() && i <= layers_above; ++i)
-    {
-        outlines_above.add(storage.getLayerOutlines(layer_idx + i, no_support, no_prime_tower));
-    }
-    outlines_above = outlines_above.unionPolygons();
+    const auto layers_below = static_cast<LayerIndex>(std::round(smooth_height / mesh.settings.get<double>("layer_height")));
```

Comment:
In the engine, we usually use `coord_t` when doing calculations, since the entire system is in those micron coords. You can use the `round_divide` utility-function we made a while back for those cases.